### PR TITLE
Add support for webpack and browser data loading

### PR DESCRIPTION
**Pull request by ehoogerbeets**
- implemented LocaleData.ensureLocale()
- now depends on updated dependencies like ilib-loader which it now loads as es6 code
    - allows webpack to see the webpack magic comments so that it knows how to bundle the locale data files
- all unit tests now work on node and on browsers with the webpack loader, including the ensureLocale() tests
- can now load data from json files as well as js files

### CODE REVIEW

File: src/DataCache.js [hunk 1]
```diff
@@ -228,6 +193,23 @@ class DataCache {
         this.logger.trace(`The data cache has been cleared.`);
         this.count = 0;
         this.data = {};
+        this.loaded.clear();
+    }
+
+    /**
+     * Record that the given file name has already been loaded.
+     */
+    fileLoaded(fileName) {
```
Comment (gouniLee) [inline src/DataCache.js:202]: IMHO, 
What about `loadFile` or `addFile` ?  actually adding is happen at the moment.
I was confused between fileLoaded and wasLoaded  at the first time.
just my thought.

File: test/testLocaleData.js [hunk 2]
```diff
@@ -1315,6 +1348,406 @@ module.exports.testLocaleData = {
...
+            test.done();
+        });
+    },
+
+    testLocaleDataEnsureLocaleDataIsCached: function(test) {
+        test.expect(14);
+        LocaleData.clearCache();
+        LocaleData.clearGlobalRoots();
+
+        LocaleData.addGlobalRoot("./test/files3");
```
[the remark this answers is not in the document]
Comment (ehoogerbeets) [reply test/testLocaleData.js:1383]: The js files are read first and it does not read the json files.

File: webpack-test.config.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ module.exports = {
             buffer: require.resolve("buffer")
         },
         alias: {
-            "calling-module": "../test/locale"
+            "calling-module": path.resolve(__dirname, "test/files3")
```
[the remark this answers is not in the document]
Comment (ehoogerbeets) [reply webpack-test.config.js:75]: Yes, exactly. It scans the test/files3 directory for any files that match the regular expression. When webpack gets to the file https://github.com/iLib-js/ilib-loader/blob/main/src/WebpackLoader.js#L112 , it reads the "magic comments" to figure out which files to include into the webpack bundle. The "calling-module" is an alias that allows the app to point to the directory where the output of ilib-assemble is.

For the locales, they come from the files in test/files3 itself. Whatever is there and matches the regular expression in the magic comment gets included.

File: src/DataCache.js (hunk 1 above)
Comment (ehoogerbeets) [reply src/DataCache.js:202]: Okay, I'll make better method names

**Review by gouniLee — APPROVED** [review]

**Pull request merged**